File: src/main/java/com/example/demo/controller/DetallePedidos/DetallePedidosController.java
package com.example.demo.controller.DetallePedidos;

import com.example.demo.Service.DetallePedidos.DetallePedidos;
import com.example.demo.Service.DetallePedidos.DetallePedidosService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import java.util.List;

@RestController
@RequestMapping("/detalles-pedidos")
public class DetallePedidosController {

    @Autowired
    private DetallePedidosService detallePedidosService;

    // GET - Obtener todos los detalles de pedidos
    @GetMapping
    public List<DetallePedidos> obtenerTodosLosDetallesPedidos() {
        return detallePedidosService.obtenerTodosLosDetallesPedidos();
    }

    // POST - Crear un nuevo detalle de pedido
    @PostMapping
    public ResponseEntity<String> crearDetallePedido(@RequestBody DetallePedidos detalle) {
        detallePedidosService.crearDetallePedido(detalle);
        return ResponseEntity.ok("Detalle de pedido " + detalle.getIdDetalle() + " creado con éxito.");
    }

    // PUT - Actualizar un detalle de pedido existente
    @PutMapping("/{id}")
    public ResponseEntity<String> editarDetallePedido(@PathVariable int id, @RequestBody DetallePedidos detalle) {
        detalle.setIdDetalle(id);
        int filas = detallePedidosService.editarDetallePedido(detalle);
        if (filas > 0) {
            return ResponseEntity.ok("Detalle de pedido con ID " + id + " actualizado correctamente.");
        } else {
            return ResponseEntity.notFound().build();
        }
    }

    // DELETE - Eliminar un detalle de pedido por ID
    @DeleteMapping("/{id}")
    public ResponseEntity<String> eliminarDetallePedido(@PathVariable int id) {
        int filas = detallePedidosService.eliminarDetallePedido(id);
        if (filas > 0) {
            return ResponseEntity.ok("Detalle de pedido con ID " + id + " eliminado correctamente.");
        } else {
            return ResponseEntity.notFound().build();
        }
    }
}

File: Frontend/app/Http/Controllers/Inventario/DetallePedidosController.php
<?php

namespace App\Http\Controllers\Inventario;

use App\Http\Controllers\Controller;
use App\Services\Inventario\DetallePedidoService;
use Illuminate\Http\Request;


class DetallePedidosController extends Controller
{
    protected $service;

    public function __construct(DetallePedidoService $service)
    {
        $this->service = $service;
    }

    /**
     * Listar todos los detalles de pedidos
     */
    public function index()
    {
        $detalles = $this->service->obtenerDetalles();

        if (!$detalles) {
            return back()->with('error', 'No se pudieron obtener los detalles de pedidos.');
        }

        return view('inventarioviews.detallepedidos.index', compact('detalles'));
    }

    /**
     * Crear un nuevo detalle de pedido
     */
    public function store(Request $request)
    {
        $data = $request->validate([
            'idPedido' => 'required|integer',
            'idProducto' => 'required|integer',
            'cantidadProducto' => 'required|integer',
            'precioUnitario' => 'required|numeric',
            'subtotal' => 'required|numeric',
        ]);

        $response = $this->service->crearDetallePedido($data);

        if (!$response['success']) {
            return back()->with('error', $response['error']);
        }

        return back()->with('success', 'Detalle de pedido creado correctamente.');
    }

    /**
     * Editar un detalle de pedido existente
     */
    public function update(Request $request, int $id)
    {
        $data = $request->validate([
            'idPedido' => 'required|integer',
            'idProducto' => 'required|integer',
            'cantidadProducto' => 'required|integer',
            'precioUnitario' => 'required|numeric',
            'subtotal' => 'required|numeric',
        ]);

        $response = $this->service->editarDetallePedido($id, $data);

        if (!$response['success']) {
            return back()->with('error', $response['error']);
        }

        return back()->with('success', 'Detalle de pedido actualizado correctamente.');
    }

    /**
     * Eliminar un detalle de pedido por ID
     */
    public function destroy(int $id)
    {
        $response = $this->service->eliminarDetallePedido($id);

        if (!$response['success']) {
            return back()->with('error', $response['error']);
        }

        return back()->with('success', 'Detalle de pedido eliminado correctamente.');
    }
}
